libs/wingc/src/type_check.rs

```diff
@@ -2372,21 +2372,26 @@ impl<'a> TypeChecker<'a> {
 						element_type = self.types.add_type(Type::Json(Some(json_data)));
 					}
 
+					// TODO: what's this?
 					// Augment the json list data with the new element type
-					if let Type::Json(Some(JsonData { ref mut kind, .. })) = &mut *element_type {
-						if let JsonDataKind::Fields(ref mut fields) = kind {
-							fields.insert(
-								sym.clone(),
-								SpannedTypeInfo {
-									type_: t,
-									span: field.span(),
-								},
-							);
-						}
-					}
-
-					self.validate_type(t, element_type, field);
+					// if let Type::Json(Some(JsonData { ref mut kind, .. })) = &mut *element_type {
+					// 	if let JsonDataKind::Fields(ref mut fields) = kind {
+					// 		fields.insert(
+					// 			key.clone(),
+					// 			SpannedTypeInfo {
+					// 				type_: t,
+					// 				span: value.span(),
+					// 			},
+					// 		);
+					// 	}
+					// }
```

Comment:
@MarkMcCulloh can you explain this? Not sure what to do here..

Reply:
This allows you to treat Map<Json> basically as a Json itself, so it keeps information about each field stored. In theory this would allow this to work implicitly:

```
struct A {
  a: str;
}

let jsonVal = Json "blah";
let x = { "a" => jsonVal };

let y: A = x;
//^ This casting works
```

However... this does not actually seem to work right now and I doubt anyone cares about it. So I'd say go ahead and delete it and it can be addressed another day

Reply:
Ok, that's what I thought it was supposed to do but it didn't work so I asked. I'll remove it for now.